examples/web/components/CodeSnippetPreview.tsx

```diff
@@ -99,23 +157,10 @@ export default function CodeSnippetPreview({
   );
   const [showLogo, setShowLogo] = useState(true);
 
-  const [preview, setPreview] = useState("");
-
-  const render = useCallback((code, lang) => {
-    getProcessor(lang).then((processor) =>
-      processor
-        .process(`\`\`\`${lang} annotate\n${code}\n\`\`\``)
-        .then((data) => {
-          setPreview(data.toString());
-        })
-    );
-  }, []);
-
-  const debouncedRender = useDebouncedCallback(render, 500);
-
-  useEffect(() => {
-    debouncedRender(code, lang);
-  }, [code, debouncedRender, lang]);
+  const deferredLang = useDeferredValue(lang);
+  const deferredCode = useDeferredValue(code);
```

Comment:
`useDeferredValue()` puts the change into low priority(transition), so concurrent mode could kick in.

for more, refer to https://jser.dev/react/2022/01/26/how-does-react-usedeferredvalue-work/